espflash/src/error.rs

```diff
@@ -19,6 +19,10 @@ use crate::{
     targets::Chip,
 };
 
+// A type alias for a dynamic error that can be used in the library.
+// https://d34dl0ck.me/rust-bites-designing-error-types-in-rust-libraries/index.html
+type CoreError = Box<dyn core::error::Error + Send + Sync>;
```

Comment:
This is a mistake, library users will not be able to inspect the error this way.

Reply:
No, because I'm wrong :) Apparently if one uses `Error::source` they can downcast the result back into the original type if they want to.